gobblin-compaction/src/main/java/gobblin/compaction/Dataset.java

```diff
@@ -118,13 +112,13 @@ public Dataset build() {
   private Path inputPath;
   private Path inputLatePath;
   private State jobProps;
+  private Optional<State> extraDatasetProps;
```

Comment:
The reason to have `extraDatasetProps` is because `jobProps` is set in JobPropsCreater, but `extraDatasetProps` is set in `DatasetsFinder`. The problem of reusing `jobProps` is that it cannot differentiate replacing a `jobProps` and adding extra properties to a `jobProps`.